hack/extract-kubeconfig.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+namespace="${1:-}"
+test="${2:-}"
+if [[ -z "${namespace}" || -z "${test}" ]]; then
+	echo "USAGE: $0 <namespace> <test>"
+	exit 1
+fi
+
+echo "Scanning ${namespace} for the \$KUBECONFIG for the $test test..."
+raw="$( oc get secrets --namespace "${namespace}" -o jsonpath="{.items[?(@.metadata.name==\"${test}\")].data.kubeconfig}" )"
+if [[ -n "${kubeconfig}" ]]; then
+	output="$( mktemp /tmp/kubeconfig.XXXXX )"
+	echo -n "${kubeconfig}" | base64 --decode > "${output}"
+	echo "\$KUBECONFIG saved to ${output}"
+	exit 0
+fi
+
+function retrieve() {
+	pods="$( oc get pods --namespace "${namespace}" --selector "ci.openshift.io/multi-stage-test=${test}" -o json )"
+	for pod in $( jq --raw-output '.items[].metadata.name' <<<"${pods}" ); do
+		kubeconfig="$( jq --raw-output ".items[] | select(.metadata.name==\"${pod}\") | .spec.containers[] | select(.name==\"test\") | .env[] | select(.name==\"KUBECONFIG\") | .value" <<<"${pods}" )"
```

Comment:
If this pod fallback is aimed at pulling the content out of a running install pod (with the earlier `get secrets` covering post-install, should we be using `install` or some such for the pod name?  How consistent are pod names?  Do we know that it's always a particular pod name providing the kubeconfig?

Reply:
oh, wait.  By the time we have `KUBECONFIG` in a pod's environment spec, we have the secret.  So is this just trying to close a race where the secret goes up right after we look for it?

Reply:
No it looks like the KUBECONFIG is in everyone's env, all the time. So we're getting it from the installing Pod before the upload to the secret.

>should we be using install or some such for the pod name?

no

> How consistent are pod names?

They aren't

> Do we know that it's always a particular pod name providing the kubeconfig?

no